### PR TITLE
Update releasing runbook

### PR DESCRIPTION
**Pull request by sdb9696**

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,9 +1,62 @@
...
+
+Or refresh an existing token
+
+This allows the generator to access github.com
+
+visit https://github.com/github-changelog-generator/github-changelog-generator#github-token
+
+```bash
+export CHANGELOG_GITHUB_TOKEN=token
+```
```
Comment (rytilahti) [inline RELEASING.md]: I'd shorten this to three simple bullet points (given this is just for maintainers anyway):
```
## Required packages
* [github client](url)
* [gitchub_changelog_generator](url)
* [github access token](url)
```

File: RELEASING.md [hunk 2]
```diff
@@ -1,9 +1,62 @@
...
+
 ## Set release information
 
+Replace 0.7.0.dev2 with intended new version number
+
```
Comment (rytilahti) [inline RELEASING.md]:
```suggestion
```

File: RELEASING.md [hunk 3]
```diff
@@ -1,9 +1,62 @@
...
+0.3.5 should always be the previous release as it's the last pyhs100 release in HISTORY.md which is the changelog prior to github release notes.
+
+```bash
+export NEW_RELEASE=0.7.0.dev2
```
Comment (rytilahti) [inline RELEASING.md]:
```suggestion
export NEW_RELEASE=0.
```

File: RELEASING.md [hunk 4]
```diff
@@ -1,9 +1,62 @@
...
+export PREVIOUS_RELEASE=0.3.5
+
```
Comment (rytilahti) [inline RELEASING.md]:
```suggestion
```

File: RELEASING.md [hunk 5]
```diff
@@ -12,76 +65,143 @@ export NEW_RELEASE=0.4.0.dev4
...
 ```
 
-## Update dependencies and run tests
+## Update dependencies
 
 ```bash
+poetry install --all-extras --sync
+```
+
+If you want to update all dependencies to latest:
```
Comment (rytilahti) [inline RELEASING.md]: Perhaps we should do this always as part of the release process? This would make sure that the newest packages are then tested to work with the release build.

File: RELEASING.md [hunk 6]
```diff
@@ -12,76 +65,143 @@ export NEW_RELEASE=0.4.0.dev4
...
+## Run pre-commit and tests
+
+```bash
+pre-commit run --all-files
+pytest kasa
+```
+
+## Create release summary
+
+Skip this step for dev releases
```
Comment (rytilahti) [inline RELEASING.md]:
```suggestion
## Create release summary (skip for dev releases)

```

File: RELEASING.md [hunk 7]
```diff
@@ -12,76 +65,143 @@ export NEW_RELEASE=0.4.0.dev4
...
+```bash
+github_changelog_generator --base HISTORY.md --user python-kasa --project python-kasa --since-tag $PREVIOUS_RELEASE --future-release $NEW_RELEASE -o CHANGELOG.md --exclude-tags-regex 'dev\d$'
 ```
 
-Remove '--exclude-tags-regex' for dev releases.
-TODO: Find a nice way to collate dev* changes to real releases. Excluding them causes them to re-appear for pre-releases, but maybe that's fine?
+You can ignore warnings about missing PR commits like below as these relate to PRs to branches other than master:
+```
+Warning: PR 908 merge commit was not found in the release branch or tagged git history and no rebased SHA comment was found
+```
```
Comment (rytilahti) [inline RELEASING.md]: Is this necessary?

File: RELEASING.md [hunk 8]
```diff
@@ -12,76 +65,143 @@ export NEW_RELEASE=0.4.0.dev4
...
-git commit --all --verbose
+export RELEASE_NOTES=$(grep -Poz '(?<=\# Changelog\n\n)(.|\n)+?(?=\#\#)' CHANGELOG.md | tr '\0' '\n' )
+echo "$RELEASE_NOTES"
+```
+
+Check the release notes output and if it doesn't look right or the command doesn't work just copy paste from CHANGELOG.md for the latest release:
+
+```bash
+export RELEASE_NOTES="paste release notes from CHANGELOG.md here"
+```
```
Comment (rytilahti) [inline RELEASING.md]:
```suggestion
```
This is so obvious that we don't need to state it for ourselves :-)

**Review by rytilahti — CHANGES_REQUESTED** [review]
Looking great! Some comments, mainly about avoiding to add unnecessary information and keeping it short as this is just for us maintainers :-)

File: RELEASING.md (hunk 5 above)
Comment (sdb9696) [reply RELEASING.md]: It feels a bit late in the day, i.e. we've been testing a bunch of code and then just before we release we update all the dependencies.  I think that should probably be a an earlier explicit PR we do in the lead up to a release.

File: RELEASING.md (hunk 7 above)
Comment (sdb9696) [reply RELEASING.md]: I spent a little bit of time looking into the warnings.  This was to avoid anyone else having to do the same.

File: RELEASING.md (hunk 8 above)
Comment (sdb9696) [reply RELEASING.md]: I don't know how cross-platform the regex is so it thought it prudent to stress to check.  Maybe I'll just add it as a comment to the `echo` bash line.

File: RELEASING.md (hunk 7 above)
Comment (rytilahti) [reply RELEASING.md]: Ok, fair enough :+1:

File: RELEASING.md (hunk 5 above)
Comment (rytilahti) [reply RELEASING.md]: I mean, it's a _prepare_ release PR after all and it just updates our own pins, so it should be fine.

File: RELEASING.md (hunk 3 above)
Comment (sdb9696) [reply RELEASING.md]: Maybe
```bash
export NEW_RELEASE=x.x.x.devx
```

File: RELEASING.md (hunk 8 above)
Comment (rytilahti) [reply RELEASING.md]: Yeah, maybe it's a good idea to print out the variables prior doing steps that change something?

File: RELEASING.md (hunk 3 above)
Comment (rytilahti) [reply RELEASING.md]: Also fine, even when it's quite obvious that this is something one should fill.

File: RELEASING.md (hunk 1 above)
Comment (sdb9696) [reply RELEASING.md]: Done

File: RELEASING.md (hunk 2 above)
Comment (sdb9696) [reply RELEASING.md]: Done

File: RELEASING.md (hunk 3 above)
Comment (sdb9696) [reply RELEASING.md]: Done

File: RELEASING.md (hunk 4 above)
Comment (sdb9696) [reply RELEASING.md]: Done

File: RELEASING.md (hunk 5 above)
Comment (sdb9696) [reply RELEASING.md]: Done

File: RELEASING.md (hunk 6 above)
Comment (sdb9696) [reply RELEASING.md]: Done

File: RELEASING.md (hunk 8 above)
Comment (sdb9696) [reply RELEASING.md]: Removed and comment added to the `echo` bash line

File: RELEASING.md [hunk 9]
```diff
@@ -12,76 +33,130 @@ export NEW_RELEASE=0.4.0.dev4
...
 
-#-exclude-tags-regex 'dev\d$'
+### For pre-release
+
+EXCLUDE_TAGS will exclude all dev tags except for the current release dev tags.
+
+Regex should be something like this `^((?!0\.7\.0)(.*dev\d))+`. The first match group negative matches on the current release and the second matches on releases ending with dev.
+
+```bash
+EXCLUDE_TAGS=${NEW_RELEASE%.dev*}; EXCLUDE_TAGS=${EXCLUDE_TAGS//"."/"\."}; EXCLUDE_TAGS="^((?!"$EXCLUDE_TAGS")(.*dev\d))+"
```
Comment (rytilahti) [inline RELEASING.md:83]:
```suggestion
export EXCLUDE_TAGS=${NEW_RELEASE%.dev*}; export EXCLUDE_TAGS=${EXCLUDE_TAGS//"."/"\."}; export EXCLUDE_TAGS="^((?!"$EXCLUDE_TAGS")(.*dev\d))+"
```
These need to be exported.

**Review by rytilahti — APPROVED** [review]
LGTM after exporting the variables.

File: RELEASING.md (hunk 9 above)
Comment (sdb9696) [reply RELEASING.md:83]: I don't think they do as they're not used by any sub processes

Comment (rytilahti) [reply RELEASING.md:83]: Ohhh, hmm, didn't think about it that way. So it's working as expected?

Comment (sdb9696) [reply RELEASING.md:83]: Yes, it works
